Dockerfile

```
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19 AS builder
+FROM golang AS builder
```

registry.access.redhat.com/ubi9/go-toolset:9.6

---

IMO we should be using public images in the open-source Dockerfiles, so I think it's fine to switch to the "upstream Go image". We can choose to build with a different image in the CI, but having the public image in the public Dockerfiles makes it much easier for the public to build the software / contribute fixes.

I would pin the image to a specific Go version though, to have an indication of what we expect to build with. In this case this would be the latest 1.24:

```suggestion
FROM docker.io/library/golang:1.24.9 AS builder
```

---

The image I suggested is publicly available.  If we wish to use the "library" version then we will need to make an additional change to the openshift CI to substitute it correctly

---

I wasn't sure if that image is available without authentication. But if it is, then that's also a possibilty. I think I would use the Go version as tag though and not the RHEL version.